azure/services/aso/aso.go

```diff
@@ -65,6 +65,8 @@ func New(ctrlClient client.Client, clusterName string) *Service {
 
 // CreateOrUpdateResource implements the logic for creating a new or updating an
 // existing resource with ASO.
+//
+//nolint:gocyclo // This function is necessarily complex.
```

Comment:
:( 

I've looked before for any low-hanging fruit to split any decent-sized chunk of this into its own function, but there's enough shared state generated within this function that I think any new helper function would end up taking a lot of random-looking parameters and return a lot of random-looking values.

Reply:
@mboersma @CecileRobertMichon Had to drop the lgtm to add this change, PTAL.